tracing/tracing_test.go

```diff
@@ -10,16 +11,63 @@ import (
 )
 
 func TestIntrumentedHTTPHandler(t *testing.T) {
-	var got *slog.Logger
+	const wantTraceID = "test-trace-id"
+	var (
+		gotLogger  *slog.Logger
+		gotTraceID string
+	)
 	handler := tracing.InstrumentHTTP(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-		got = slog.FromCtx(req.Context())
+		gotLogger = slog.FromCtx(req.Context())
+		gotTraceID, _ = tracing.CtxGetTraceID(req.Context())
 	}))
 
 	req := httptest.NewRequest(http.MethodGet, "/", nil)
+	req.Header.Set("traceparent", wantTraceID)
 	res := httptest.NewRecorder()
+
 	handler.ServeHTTP(res, req)
 
-	if got == nil {
+	if gotLogger == nil {
 		t.Fatal("got nil logger")
 	}
+
+	if gotTraceID != wantTraceID {
+		t.Fatalf("got %q != want %q", gotTraceID, wantTraceID)
+	}
+}
+
+func TestCtxWithTraceAndOrgID(t *testing.T) {
+	const (
+		wantTraceID = "trace-id-value"
+		wantOrgID   = "org-id-value"
+	)
+	ctx := context.Background()
+
+	got, ok := tracing.CtxGetTraceID(ctx)
+	if ok {
+		t.Fatalf("unexpected trace id: %q", got)
+	}
+	got, ok = tracing.CtxGetOrgID(ctx)
+	if ok {
+		t.Fatalf("unexpected trace id: %q", got)
+	}
+
+	ctx = tracing.CtxWithTraceID(ctx, wantTraceID)
+	ctx = tracing.CtxWithOrgID(ctx, wantOrgID)
```

Comment:
Testing each isolated would make it harder to catch bugs like using the same key for both, then only one of them works at a time, wanted to be sure that using both together works fine since they are the only general purpose metadata we are going to have on our systems.

Still there is no way to extract org ID automatically since we don't have a header for that, code will have to do that manually for now since it comes on request bodies.